src/Ilios/CliBundle/Tests/Command/AddNewStudentsToSchoolCommandTest.php

```diff
@@ -0,0 +1,199 @@
+            ->andReturn(['abc2']);
+            
+        $this->userManager->shouldReceive('createUser')->andReturn($user);
+        $this->userManager
+            ->shouldReceive('updateUser')
+            ->with($user)->once();
+        $this->schoolManager->shouldReceive('findSchoolBy')->with(array('id' => 1))->andReturn($school);
+        $role = m::mock('Ilios\CoreBundle\Entity\UserRoleInterface')
+            ->shouldReceive('addUser')->with($user)
+            ->mock();
+        $user->shouldReceive('addRole')->with($role);
+        $this->userRoleManager
+            ->shouldReceive('findUserRoleBy')
+            ->with(array('title' => 'Student'))
+            ->andReturn($role);
+        $this->userRoleManager
+            ->shouldReceive('updateUserRole')
+            ->with($role);
+
+        $this->authenticationManager->shouldReceive('createAuthentication')->andReturn($authentication);
+        $this->authenticationManager->shouldReceive('updateAuthentication')->with($authentication, false);
+        
+        $this->sayYesWhenAsked();
+        $this->commandTester->execute(array(
+            'command'   => self::COMMAND_NAME,
+            'schoolId'    => 1,
+            'filter'    => 'FILTER',
+        ));
+        
+        $output = $this->commandTester->getDisplay();
+        $this->assertRegExp(
+            '/Found 2 students in the directory./',
+            $output
+        );
+        
+        $this->assertRegExp(
+            '/There are 1 new students to be added to school 1./',
+            $output
+        );
+        $this->assertRegExp(
+            '/Do you wish to add these students to school 1?/',
+            $output
+        );
+        $this->assertRegExp(
+            '/abc       \| first \| last \| email /',
+            $output
+        );
+        $this->assertRegExp(
+            '/Success! New student #42 first last created./',
+            $output
+        );
+    }
+    
+    public function testBadSchoolId()
+    {
+        $this->schoolManager->shouldReceive('findSchoolBy')->with(array('id' => 1))->andReturn(null);
+        $this->setExpectedException('Exception', 'School with id 1 could not be found.');
+        $this->commandTester->execute(array(
+            'command'      => self::COMMAND_NAME,
+            'filter'         => 'FILTER',
+            'schoolId'         => '1'
+        ));
+        
+    }
+    
+    public function testFilterRequired()
+    {
+        $this->setExpectedException('RuntimeException');
+        $this->commandTester->execute(array(
+            'command'      => self::COMMAND_NAME,
+            'schoolId'         => '1'
+        ));
+    }
+    
+    public function testSchoolRequired()
+    {
+        $this->setExpectedException('RuntimeException');
+        $this->commandTester->execute(array(
+            'command'      => self::COMMAND_NAME,
+            'filter'         => '1',
+        ));
+    }
+    
+
+    protected function sayYesWhenAsked()
```

Comment:
I stole this: http://symfony.com/doc/current/components/console/helpers/dialoghelper.html#testing-a-command-which-expects-input  I actually thought maybe I had written more of it than that, but pretty much just named the function.